Sources/XcbeautifyLib/Parser.swift

```diff
@@ -15,86 +15,86 @@ public class Parser {
-        innerParser(ParallelTestingStartedCaptureGroup.regex, outputType: .test),
-        innerParser(ParallelTestingPassedCaptureGroup.regex, outputType: .test),
-        innerParser(ParallelTestSuiteStartedCaptureGroup.regex, outputType: .test),
-        innerParser(LDWarningCaptureGroup.regex, outputType: .warning),
-        innerParser(CompileWarningCaptureGroup.regex, outputType: .warning),
-        innerParser(GenericWarningCaptureGroup.regex, outputType: .warning),
-        innerParser(WillNotBeCodeSignedCaptureGroup.regex, outputType: .warning),
-        innerParser(ClangErrorCaptureGroup.regex, outputType: .error),
-        innerParser(CheckDependenciesErrorsCaptureGroup.regex, outputType: .error),
-        innerParser(ProvisioningProfileRequiredCaptureGroup.regex, outputType: .error),
-        innerParser(NoCertificateCaptureGroup.regex, outputType: .warning),
-        innerParser(FileMissingErrorCaptureGroup.regex, outputType: .error),
-        innerParser(XcodebuildErrorCaptureGroup.regex, outputType: .error),
-        innerParser(CompileErrorCaptureGroup.regex, outputType: .error),
-        innerParser(CursorCaptureGroup.regex, outputType: .warning),
-        innerParser(FatalErrorCaptureGroup.regex, outputType: .error),
-        innerParser(LDErrorCaptureGroup.regex, outputType: .error),
-        innerParser(LinkerDuplicateSymbolsLocationCaptureGroup.regex, outputType: .error),
-        innerParser(LinkerDuplicateSymbolsCaptureGroup.regex, outputType: .error),
-        innerParser(LinkerUndefinedSymbolLocationCaptureGroup.regex, outputType: .error),
-        innerParser(LinkerUndefinedSymbolsCaptureGroup.regex, outputType: .error),
-        innerParser(PodsErrorCaptureGroup.regex, outputType: .error),
-        innerParser(SymbolReferencedFromCaptureGroup.regex, outputType: .error),
-        innerParser(ModuleIncludesErrorCaptureGroup.regex, outputType: .error),
-        innerParser(ParallelTestingFailedCaptureGroup.regex, outputType: .nonContextualError),
-        innerParser(ParallelTestCaseFailedCaptureGroup.regex, outputType: .error),
-        innerParser(ShellCommandCaptureGroup.regex, outputType: .task),
-        innerParser(UndefinedSymbolLocationCaptureGroup.regex, outputType: .warning),
-        innerParser(PackageFetchingCaptureGroup.regex, outputType: .task),
-        innerParser(PackageUpdatingCaptureGroup.regex, outputType: .task),
-        innerParser(PackageCheckingOutCaptureGroup.regex, outputType: .task),
-        innerParser(PackageGraphResolvingStartCaptureGroup.regex, outputType: .task),
-        innerParser(PackageGraphResolvingEndedCaptureGroup.regex, outputType: .task),
-        innerParser(PackageGraphResolvedItemCaptureGroup.regex, outputType: .task),
-        innerParser(DuplicateLocalizedStringKeyCaptureGroup.regex, outputType: .warning)
+        innerParser(AnalyzeCaptureGroup.regex, outputType: AnalyzeCaptureGroup.outputType),
```

Comment:
What do you think about doing the following?

```swift
func innerParser<T: CaptureGroup>(captureGroup: T) -> ... {
 // Read .regex and `.outputType
}
```

That way you eliminate the root for inconsistency between the first and the second argument, which are both reading static variables on the same type.

Reply:
Nevermind. I saw that you did that [here](https://github.com/tuist/xcbeautify/pull/177)